### PR TITLE
Add bastion tables (#435)

### PR DESCRIPTION
**Pull request by misraved**
Co-authored-by: Scott Fletcher <scott.fletcher@oracle.com>
Co-authored-by: Ved misra <47312748+misraved@users.noreply.github.com>

This PR includes the following tables:
- oci_bastion_bastion
- oci_bastion_session

# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
Add example SQL query results here (please include the input queries as well)
```
</details>

### CODE REVIEW

File: oci/table_oci_bastion_session.go [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+			request.Page = response.OpcNextPage
+		} else {
+			pagesLeft = false
+		}
+	}
+
+	return nil, err
+}
+
+// // HYDRATE FUNCTION
```
Comment (ParthaI) [inline oci/table_oci_bastion_session.go:208]:
```suggestion
//// HYDRATE FUNCTION
```

File: oci/table_oci_bastion_session.go [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+
+	var id string
+	if h.Item != nil {
+		id = *h.Item.(bastion.SessionSummary).Id
+	} else {
+		id = d.KeyColumnQuals["id"].GetStringValue()
+	}
+
+	// handle empty id in get call
```
Comment (ParthaI) [inline oci/table_oci_bastion_session.go:220]:
```suggestion
	// Handle empty id in get call
```

**Review by ParthaI — COMMENTED** [review]
@misraved please see the comments, Thanks!
It would be great if we could add more example queries for the tables.

For Bastion table using the following columns e.g target_vcn_id, target_subnet_id, lifecycle_state etc.

For Session table using following columns e.g bastion_id(join query), key_details, key_type, lifecycle_state etc.

File: oci/service.go [hunk 3]
```diff
@@ -1446,6 +1448,48 @@ func analyticsService(ctx context.Context, d *plugin.QueryData, region string) (
 	return sess, nil
 }
 
+// bastionService returns the service client for OCI Bastion service
+func bastionService(ctx context.Context, d *plugin.QueryData, region string) (*session, error) {
+	logger := plugin.Logger(ctx)
+
+	// have we already created and cached the service?
```
Comment (ParthaI) [inline oci/service.go:1455]:
```suggestion
	// Have we already created and cached the service?
```

File: oci/service.go [hunk 4]
```diff
@@ -1446,6 +1448,48 @@ func analyticsService(ctx context.Context, d *plugin.QueryData, region string) (
...
+	serviceCacheKey := fmt.Sprintf("bastion-%s", region)
+	if cachedData, ok := d.ConnectionManager.Cache.Get(serviceCacheKey); ok {
+		return cachedData.(*session), nil
+	}
+
+	// get oci config info from steampipe connection
```
Comment (ParthaI) [inline oci/service.go:1461]:
```suggestion
	// Get OCI config info from steampipe connection
```

File: oci/service.go [hunk 5]
```diff
@@ -1446,6 +1448,48 @@ func analyticsService(ctx context.Context, d *plugin.QueryData, region string) (
...
+	ociConfig := GetConfig(d.Connection)
+
+	provider, err := getProvider(ctx, d.ConnectionManager, region, ociConfig)
+	if err != nil {
+		logger.Error("bastionService", "getProvider.Error", err)
+		return nil, err
+	}
+
+	// get bastion service client
```
Comment (ParthaI) [inline oci/service.go:1470]:
```suggestion
	// Get bastion service client
```

File: oci/service.go [hunk 6]
```diff
@@ -1446,6 +1448,48 @@ func analyticsService(ctx context.Context, d *plugin.QueryData, region string) (
...
+	client, err := bastion.NewBastionClientWithConfigurationProvider(provider)
+	if err != nil {
+		return nil, err
+	}
+
+	// get tenant ocid from provider
```
Comment (ParthaI) [inline oci/service.go:1476]:
```suggestion
	// Get tenant ocid from provider
```

File: oci/service.go [hunk 7]
```diff
@@ -1446,6 +1448,48 @@ func analyticsService(ctx context.Context, d *plugin.QueryData, region string) (
...
+	if err != nil {
+		return nil, err
+	}
+
+	sess := &session{
+		TenancyID:       tenantId,
+		BastionClient:   client,
+	}
+
+	// save session in cache
```
Comment (ParthaI) [inline oci/service.go:1487]:
```suggestion
	// Save session in cache
```

**Review by ParthaI — COMMENTED** [review]
Minor comments

File: oci/service.go (hunk 3 above)
Comment (misraved) [reply oci/service.go:1455]: @ParthaI I believe we are just following the pattern from other services.

File: oci/service.go (hunk 4 above)
Comment (misraved) [reply oci/service.go:1461]: @ParthaI I believe we are just following the pattern from other services.

File: oci/service.go (hunk 5 above)
Comment (misraved) [reply oci/service.go:1470]: @ParthaI I believe we are just following the pattern from other services.

**Review by ParthaI — APPROVED** [review]

**Pull request merged**